pkg/kcp/provider/azure/vpcpeering/deleteVpcPeering.go

```diff
@@ -4,34 +4,41 @@ import (
 	"context"
 	"github.com/kyma-project/cloud-manager/pkg/composed"
 	azuremeta "github.com/kyma-project/cloud-manager/pkg/kcp/provider/azure/meta"
+	"github.com/kyma-project/cloud-manager/pkg/kcp/provider/azure/util"
+	"k8s.io/utils/ptr"
 )
 
 func deleteVpcPeering(ctx context.Context, st composed.State) (error, context.Context) {
 	state := st.(*State)
 	logger := composed.LoggerFromCtx(ctx)
 	obj := state.ObjAsVpcPeering()
 
-	if len(obj.Status.Id) == 0 {
-		logger.Info("VpcPeering deleted before Azure peering is created")
+	if state.localPeering == nil {
+		logger.Info("Skipping local peering deletion")
 		return nil, nil
 	}
 
-	resourceGroupName := state.Scope().Spec.Scope.Azure.VpcNetwork
+	resourceId, err := util.ParseResourceID(ptr.Deref(state.localPeering.ID, ""))
+
+	if err != nil {
+		logger.Error(err, "Failed parsing localPeering.ID while deleting local peering")
+		return nil, nil
+	}
 
 	logger.Info("Deleting VpcPeering")
 
-	err := state.localClient.DeletePeering(
+	err = state.localClient.DeletePeering(
 		ctx,
-		resourceGroupName,
-		state.Scope().Spec.Scope.Azure.VpcNetwork,
+		resourceId.ResourceGroup,
+		resourceId.ResourceName,
 		obj.GetLocalPeeringName(),
 	)
 
 	if err != nil {
-		return azuremeta.LogErrorAndReturn(err, "Error deleting vpc peering", ctx)
+		return azuremeta.LogErrorAndReturn(err, "Error deleting local vpc peering", ctx)
```

Comment:
This azuremeta.LogErrorAndReturn is hiding behavior.
Consider not using it for return values (outside of this PR)